core/DataAccess/Model.php

```diff
@@ -786,10 +788,35 @@ public function isArchiveAlreadyInProgress($invalidatedArchive)
             $bind[] = $invalidatedArchive['report'];
         }
 
-        $sql = "SELECT MAX(idinvalidation) FROM `$table` WHERE idsite = ? AND date1 = ? AND date2 = ? AND `period` = ? AND `name` = ? AND status = 1 AND $reportClause";
+        $sql = "SELECT MAX(idinvalidation) as idinvalidation, ts_invalidated FROM `$table` WHERE idsite = ? AND date1 = ? AND date2 = ? AND `period` = ? AND `name` = ? AND status = 1 AND $reportClause";
 
-        $inProgressInvalidation = Db::fetchOne($sql, $bind);
-        return $inProgressInvalidation;
+        $inProgressInvalidation = Db::fetchRow($sql, $bind);
+        if (empty($inProgressInvalidation)) {
+            return false;
+        }
+
+        // double check the invalidation time in case the job failed
+        try {
+            $tsInvalidated = Date::factory($inProgressInvalidation['ts_invalidated']);
+        } catch (\Exception $ex) {
+            return false; // sanity check in case stored date is broken
+        }
+
+        // if it's been a day since the archive was created, we just assume the job failed
+        $lockExpireTime = $tsInvalidated->getTimestamp() + 24 * 60 * 60;
+        if (time() < $lockExpireTime) {
+            return true;
+        }
+
+        StaticContainer::get(LoggerInterface::class)->info("Found invalidation over a day old (ts_invalidated = {inv}, idinvalidation = {idinv}), assuming it failed and deleting.", [
+            'inv' => $inProgressInvalidation['ts_invalidated'],
+            'idinv' => $inProgressInvalidation['idinvalidation'],
+        ]);
+
+        // delete the in progress invalidation since we assume it errored
+        $this->deleteInvalidations($inProgressInvalidation);
```

Comment:
random thought... might not make any sense. If we assume it failed, would we rather set it to `0` so it will be rearchived again in the hope it works and doesn't fail the next time? Although I suppose this can get complicated as we'd probably then also would maybe need to invalidate any parent period in case they were finished in the meantime?

Reply:
Parent periods shouldn't start if an intersecting lower period is there so this *should* be ok. The period might just keep failing though.

Reply:
Not sure what to do best here. I'm tempted to say we set the status `0` and better maybe fail repeatedly than having potentially wrong data? Problem obviously being it wouldn't archive any other data until this is resolved although I think if newer invalidations are added, and they are for newer periods, then these ones would be maybe processed eventually because we prefer more recent dates AFAIK. Might not be thinking this fully through though.

Reply:
@tsteur Actually, this is in the check for an existing in progress archive, so that would mean there already is another invalidation identical to this one.

I guess we can also check in CoreArchive.php for in progress invalidations that are over a day old and change the status on them before we actually start any archiving. What do you think about that?

Reply:
I see. I guess here in `isArchiveAlreadyInProgress ` if the same invalidation wants to start, and there is one in progress for 1 day already then we can delete it.

There would be still some that would never be detected. It could be in CronArchive.php or could be a task to check for these? Could in this case maybe only use this logic and undo the change in `isArchiveAlreadyInProgress ` as having a general check should help? Ideally we just don't execute the query looking for in progress archive invalidations older than 1 day too often. Maybe only once hourly or once on cronarchive start or so.

Reply:
> There would be still some that would never be detected. It could be in CronArchive.php or could be a task to check for these? Could in this case maybe only use this logic and undo the change in isArchiveAlreadyInProgress as having a general check should help? Ideally we just don't execute the query looking for in progress archive invalidations older than 1 day too often. Maybe only once hourly or once on cronarchive start or so.

Yes, this was what I mentioned in the second paragraph. I'll go w/ this.